src/Control/MonadZero.purs

```diff
@@ -39,15 +39,22 @@ instance monadZeroArray :: MonadZero Array
 -- | For example:
 -- |
 -- | ```purescript
--- | import Data.Array
+-- | import Prelude
+-- | import Control.Monad (bind)
+-- | import Control.MonadZero (guard)
+-- | import Data.Array ((..))
 -- |
--- | factors :: Number -> Array Number
+-- | factors :: Int -> Array Int
 -- | factors n = do
--- |   a <- 1 .. n
--- |   b <- 1 .. a
+-- |   a <- 1..n
+-- |   b <- a..n
 -- |   guard $ a * b == n
--- |   pure a
+-- |   pure [a, b]
 -- | ```
+-- | In this example, a step of the array comprehension will fail with
```

Comment:
I wouldn't mention `empty` here, since readers of the documentation won't necessarily be reading the source code.